docs/source/weightvec.rst

```diff
@@ -47,16 +47,84 @@ Let ``w`` be an instance of ``WeightVec``:
 
 .. function:: sum(w)
 
-    Get the sum of weights. 
+    Get the sum of weights.
 
     :note: The sum of weights is maintained by the weight vector, and thus this function can immediately return the value in ``O(1)`` (without computation).
 
 
-Why we want a WeightVec type
------------------------------
+Why we want an AbstractWeights type
```

Comment:
As you like. You can't fix all typos in the code, but this line is quite close to the ones you touched, so...